Wrappers/Python/ccpi/web_viewer/trame_viewer.py

```diff
@@ -0,0 +1,772 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import os
+
+import matplotlib.pyplot as plt
+from trame import update_layout
+from trame.html import vtk, vuetify
+from trame.layouts import SinglePageWithDrawer
+from vtkmodules.util import colors
+from vtkmodules.vtkIOImage import vtkMetaImageReader
+
+from ccpi.viewer.CILViewer import CILViewer
+from ccpi.viewer.CILViewer2D import SLICE_ORIENTATION_XY, SLICE_ORIENTATION_XZ, SLICE_ORIENTATION_YZ
+from ccpi.viewer.utils.conversion import cilHDF5ResampleReader
+from ccpi.web_viewer.camera_data import CameraData
+
+DEFAULT_SLICE = 32
+INITIAL_IMAGE = "head.mha"
+
+
+class TrameViewer:
+    def __init__(self, list_of_files=None):
+        # Define attributes that will be constructed in methods outside of __init__
+        self.cmin = None
+        self.cmax = None
+        self.windowing_defaults = None
+        self.slice_window_range_defaults = None
+        self.slice_level_default = None
+        self.slice_window_default = None
+        self.max_slice = None
+        self.default_slice = None
+        self.image = None
+        self.slice_interaction_col = None
+        self.slice_interaction_row = None
+        self.slice_interaction_section = None
+        self.volume_interaction_col = None
+        self.volume_interaction_row = None
+        self.volume_interaction_section = None
+        self.windowing_slider_is_percentage = False  # Defaults to not percentage with the head.mha file
+        self.colour_slider_is_percentage = False  # Defaults to not percentage with the head.mha file
+        self.slice_level_slider_is_percentage = False  # Defaults to not percentage with the head.mha file
+        self.slice_window_slider_is_percentage = False  # Defaults to not percentage with the head.mha file
+        self.slice_window_sliders_are_detailed = False  # Defaults to none-detailed sliders
+        self.disable_2d = False
+        self.disable_3d = False
+
+        # Define UI elements in __init__ to quiet down Pep8
+        self.model_choice = None
+        self.background_choice = None
+        self.toggle_slice_visibility = None
+        self.slice_slider = None
+        self.toggle_window_details_button = None
+        self.orientation_radio_buttons = None
+        self.slice_window_range_slider = None
+        self.slice_window_slider = None
+        self.slice_level_slider = None
+        self.toggle_volume_visibility = None
+        self.opacity_radio_buttons = None
+        self.colour_choice = None
+        self.clipping_switch = None
+        self.colour_slider = None
+        self.windowing_range_slider = None
+        self.reset_cam_button = None
+        self.reset_defaults_button = None
+
+        # Load files and setup the CILViewer
+        if list_of_files is None:
+            self.list_of_files = os.listdir("data/")
```

Comment:
I like this because it solves the issue of forgetting to pass an arg to the program but it makes sense to raise an exception here probably, I will change it.